.travis.yml

```diff
@@ -21,6 +21,11 @@ cache:
 before_install:
   - npm install -g npm@5.5.1
   - sudo apt-get install libgconf2-4
+  - export SOURCE_BRANCH=${TRAVIS_PULL_REQUEST_BRANCH:-$TRAVIS_BRANCH}
+  - export IS_FORKED_PULL_REQUEST=$([[ $TRAVIS_PULL_REQUEST != false && $TRAVIS_PULL_REQUEST_SLUG != $TRAVIS_REPO_SLUG ]] && echo true || echo false)
+  - export IS_BETA=$([[ $TRAVIS_TAG =~ .*beta.* && $IS_FORKED_PULL_REQUEST = false ]] && echo true || echo false)
+  - export IS_RELEASE=$([[ $SOURCE_BRANCH =~ ^.*release.*$|^master$ && $IS_FORKED_PULL_REQUEST = false ]] && echo true || echo false)
```

Comment:
I think we can remove `^master$` from the source branch regex, since our releases occur on release branches? Or is this for the nightly build? If so, I would prefer to create a separate variable `IS_NIGHTLY`. I think it will be helpful to distinguish this release from an official release.

For the official release branches, the format, is `<year>-<month>-release`. I think we can mark `release` as the end of the regex. We could also be more strict if the regex was looking for `4 numbers|dash|some lowercase letters|dash|release`.
e.g. `^[0-9]{4}-[a-z]+-release$`.

But we could keep it as just `ends with release` too.

Reply:
Yup, I'll remove `^master$`. Additionally, I realized that the requirement for a commit to count as a release isn't the name of the branch, but rather the tag given to the commit. I'll change the condition.

For nightly builds, they correspond with `IS_BETA`, since betas are tagged every day.

As for the format, I agree and I'll make it more strict.